paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -605,6 +605,14 @@ internal abstract class BaseSheetViewModel(
             }
         )
 
+        if (mostRecentlySelectedSavedPaymentMethod.value?.id == paymentMethodId) {
```

Comment:
The change to `SAVED_PM_SELECTION` doesn't, because tab mode doesn't use it.

The change to `SAVE_SELECTION` theoretically does -- but in practice, the behavior is the same. In tab mode, we decide which payment methods to show based on `paymentMethods`. When you remove a PM, it is removed from `paymentMethods` and it is no longer displayed. Currently, even though the PM isn't displayed and you can't pay with it, it is still the payment selection. This change makes it so that it's not the selection anymore, but that doesn't actually impact what the user could do.

We have to make this change for vertical mode, because vertical mode is using the current selection to determine what to display, rather than using `paymentMethods`.